freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/WorkloadCredential.java

```diff
@@ -32,4 +33,12 @@ public String getHashedPassword() {
     public Optional<Instant> getExpirationDate() {
         return expirationDate;
     }
+
+    public static WorkloadCredential fromGetActorWorkloadCredentialsResponse(GetActorWorkloadCredentialsResponse response) {
```

Comment:
few things:
- method name should start with verb, like create
- I don't like this static method, because it will make it hard to mock during testing. Maybe moving this logic into a converter component would make it easier to mock and test
- missing tests

Reply:
Fixed. Also, instead of moving to a provider I ended up put the logic in a UmsCredentialProvider which also makes the grpc call.